pkgs/development/tools/misc/cc-tool/default.nix

```diff
@@ -0,0 +1,36 @@
+{ stdenv
+, fetchFromGitHub
+, autoreconofHook
+, boost
+, libusb1
+}:
+
+stdenv.mkDerivation rec {
+  pname = "cc-tool";
+  version = "unstable-2020-05-19";
+
+  src = fetchFromGitHub {
+    owner = "dashesy";
+    repo = pname;
+    rev = "19e707eafaaddee8b996ad27a9f3e1aafcb900d2";
+    hash = "sha256:1f78j498fdd36xbci57jkgh25gq14g3b6xmp76imdpar0jkpyljv";
+  };
+
+  nativeBuildInputs = [ pkg-config autoreconfHook ];
+  buildInputs = [ libusb1 boost ];
+
+  postInstall = ''
+    install -D udev/90-cc-debugger.rules $out/lib/udev/rules.d/90-cc-debugger.rules
+  '';
+
+  meta = with stdenv.lib; {
+    description = "Command line tool for the Texas Instruments CC Debugger";
+    longDescription = ''
+      cc-tool provides support for Texas Instruments CC Debugger
+    '';
+    homepage = "https://github.com/dashesy/cc-tool";
+    license = licenses.gpl2;
+    platforms = with platforms; linux;
```

Comment:
Does this work on macOS? If you want, I can try.

Reply:
I do not have a macOS system available. I've added it to the available platforms, so it would be great if you can test it.
If no CC debugger is available, it will print:
```
  CC Debugger device not found
```

Reply:
It compiles fine on macOS :+1: . This looks almost good to go. My only worry is that the udev rules make the device world-writable, which is not really nice from a security perspective. I am not sure how serious a thread this is, but I guess it could be used by another account to reflash a device? One option would be to do something like the following change, which only makes the device only accessible to root and members of `plugdev`:

```diff
diff --git a/pkgs/development/tools/misc/cc-tool/default.nix b/pkgs/development/tools/misc/cc-tool/default.nix
index fa9f7b10c09..6be0ab2acb5 100644
--- a/pkgs/development/tools/misc/cc-tool/default.nix
+++ b/pkgs/development/tools/misc/cc-tool/default.nix
@@ -20,6 +20,11 @@ stdenv.mkDerivation rec {
   nativeBuildInputs = [ autoreconfHook pkg-config ];
   buildInputs = [ boost libusb1 ];
 
+  postPatch = ''
+    substituteInPlace udev/90-cc-debugger.rules \
+      --replace 'MODE="0666"' 'MODE="0660", GROUP="plugdev", TAG+="uaccess"'
+  '';
+
   postInstall = ''
     install -D udev/90-cc-debugger.rules $out/lib/udev/rules.d/90-cc-debugger.rules
   '';
```

Any opinions? (I couldn't test the above, since I don't have such a device.)